recipes-kernel/linux/files/bcm2712-raspberrypi5-domd.dts

```diff
@@ -0,0 +1,287 @@
+		};
+
+		macb_hclk: macb_hclk {
+			compatible = "fixed-clock";
+			#clock-cells = <0>;
+			clock-output-names = "hclk";
+			clock-frequency = <200000000>;
+		};
+
+		 sd_io_1v8_reg: sd_io_1v8_reg {
+			compatible = "regulator-gpio";
+			states = <0x1b7740 0x1 0x325aa0 0x0>;
+			gpios = <&gpio7d517c00 0x3 0x0>;
+			regulator-settling-time-us = <0x1388>;
+			regulator-always-on;
+			regulator-boot-on;
+			regulator-max-microvolt = <0x325aa0>;
+			regulator-min-microvolt = <0x1b7740>;
+			regulator-name = "vdd-sd-io";
+			extracted,path = "/sd_io_1v8_reg";
+			status = "okay";
+		};
+
+		gpio7d517c00: gpio@107d517c00 {
+			compatible = "brcm,brcmstb-gpio";
+			#gpio-cells = <0x2>;
+			gpio-controller;
+			reg = <0x10 0x7d517c00 0x00 0x40>;
+			brcm,gpio-direct;
+			brcm,gpio-bank-widths = <0x11 0x6>;
+			xen,path = "/soc/gpio@7d517c00";
+			xen,force-assign-without-iommu = <0x1>;
+			xen,reg = <0x10 0x7d517000 0x0 0x1000 0x10 0x7d517000>;
+		};
+
+		sd_vcc_reg: sd_vcc_reg {
+			compatible = "regulator-fixed";
+			gpios = <&gpio7d517c00 0x4 0x0>;
+			enable-active-high;
+			regulator-boot-on;
+			regulator-max-microvolt = <0x325aa0>;
+			regulator-min-microvolt = <0x325aa0>;
+			regulator-name = "vcc-sd";
+			extracted,path = "/sd_vcc_reg";
+			status = "okay";
+		};
+
+		mip0: msi-controller@1000130000 {
+			compatible = "brcm,bcm2712-mip-intc";
+			reg = <0x10 0x00130000 0x0 0xc0>;
+			interrupt-parent = <&gic>;
+			msi-controller;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			brcm,msi-base-spi = <128>;
+			brcm,msi-num-spis = <64>;
+			brcm,msi-offset = <0>;
+			brcm,msi-pci-addr = <0xff 0xfffff000>;
+			xen,path = "/axi/msi-controller@130000";
+			xen,force-assign-without-iommu = <0x1>;
+			xen,reg = <0x10 0x00130000 0x0 0x1000 0x10 0x00130000>;
+			interrupts = <0x00 0x80 0x04 0x00 0x81 0x04 0x00 0x82 0x04
+				      0x00 0x83 0x04 0x00 0x84 0x04 0x00 0x85 0x04
+				      0x00 0x86 0x04 0x00 0x87 0x04 0x00 0x88 0x04
+				      0x00 0x89 0x04 0x00 0x8a 0x04 0x00 0x8b 0x04
+				      0x00 0x8c 0x04 0x00 0x8d 0x04 0x00 0x8e 0x04
+				      0x00 0x8f 0x04 0x00 0x90 0x04 0x00 0x91 0x04
+				      0x00 0x92 0x04 0x00 0x93 0x04 0x00 0x94 0x04
+				      0x00 0x95 0x04 0x00 0x96 0x04 0x00 0x97 0x04
+				      0x00 0x98 0x04 0x00 0x99 0x04 0x00 0x9a 0x04
+				      0x00 0x9b 0x04 0x00 0x9c 0x04 0x00 0x9d 0x04
+				      0x00 0x9e 0x04 0x00 0x9f 0x04 0x00 0xa0 0x04>;
+		};
+
+		pciex: pcie@1000120000 {
+			compatible = "brcm,bcm2712-pcie";
+			#address-cells = <0x3>;
+			#size-cells = <0x2>;
+			#interrupt-cells = <0x1>;
+			reg = <0x10 0x120000 0x0 0x9310>;
+			aspm-no-l0s;
+			brcm,vdm-qos-map = <0xbbaa9888>;
+			brcm,enable-mps-rcb;
+			brcm,enable-l1ss;
+			dma-ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0x400000
+				      0x43000000 0x10 0x0 0x0 0x0 0x10 0x0>;
+			ranges = <0x2000000 0x0 0x0 0x1f 0x0 0x0 0xfffffffc
+				  0x43000000 0x4 0x0 0x1c 0x0 0x3 0x0>;
+			interrupt-parent = <&gic>;
+//			msi-parent = <&mip0>;
```

Comment:
It deserves a comment then. Something like "HACK: Actual MSI parent is &mip but due to.... we temporary use different parent"

Reply:
You can  get description from: https://github.com/xen-troops/meta-xt-rpi5/wiki/RPI-5-Xen-Dom0less-boot-exercise#pci2-msi-is-not-xen-compatible

```
"HACK: But the mip0: msi-controller@130000 doesn't have IRQs defined in the standard way and instead uses custom properties: brcm,msi-base-spi to define GIC IRQ offset, and brcm,msi-num-spis to define number of GIC IRQs to use. The Xen doesn't understand this and so can't map interrupts to DomU in case of 'xen,passthrough'.
The attempt to add correct interrupts property to the mip0: msi-controller@130000 has failed - no IRQs received by macb 1f00100000.ethernet. Reason unknown."

```

Reply:
Comment added